extensions/common/json-ld/src/main/java/org/eclipse/edc/jsonld/transformer/TransformerUtil.java

```diff
@@ -0,0 +1,124 @@
+            var jsonString = ((JsonObject) value).getJsonString(VALUE);
+            transformString(jsonString, resultFunction, context);
+        } else if (value instanceof JsonArray) {
+            transformString(((JsonArray) value).get(0), resultFunction, context);
+        } else {
+            context.reportProblem(format("Invalid property. Expected JsonString but got %s",
+                    value.getClass().getSimpleName()));
+        }
+    }
+    
+    /**
+     * Transforms a JsonValue to the desired output type and applies the result function. If the
+     * value parameter is not of type JsonObject, a problem is reported to the context.
+     *
+     * @param value the value to transform
+     * @param type the desired result type
+     * @param resultFunction the function to apply to the transformation result
+     * @param context the transformer context
+     * @param <T> the desired result type
+     */
+    public static <T> void transformObject(JsonValue value, Class<T> type, Consumer<T> resultFunction, TransformerContext context) {
+        if (value instanceof JsonObject) {
+            var result = context.transform((JsonObject) value, type);
+            resultFunction.accept(result);
+        } else if (value instanceof JsonArray) {
+            var array = (JsonArray) value;
+            transformObject(array.get(0), type, resultFunction, context);
+        } else {
+            context.reportProblem(format("Invalid property of type %s. Expected JsonObject but got %s",
+                    type.getSimpleName(), value.getClass().getSimpleName()));
+        }
+    }
+    
+    /**
+     * Transforms a JsonValue to the desired output type. The result can be a single instance or a
+     * list of that type, depending on whether the given value is a JsonObject or a JsonArray. If
+     * the result is a single instance, the result function is applied. If it is a list, the list
+     * result function is applied. If the value parameter is neither of type JsonObject nor
+     * JsonArray, a problem is reported to the context.
+     *
+     * @param value the value to transform
+     * @param type the desired result type
+     * @param resultFunction the function to apply to the transformation result, if it is a single instance
+     * @param listResultFunction the function to apply to the transformation result, if it is a list
+     * @param context the transformer context
+     * @param <T> the desired result type
+     */
+    public static <T> void transformArrayOrObject(JsonValue value, Class<T> type, Consumer<T> resultFunction,
+                                                  Consumer<List<T>> listResultFunction, TransformerContext context) {
+        if (value instanceof JsonArray) {
+            var jsonArray = (JsonArray) value;
+            var results = jsonArray.stream().map(entry -> context.transform(entry, type)).collect(toList());
+            listResultFunction.accept(results);
```

Comment:
having two separate arguments for `list` and `single` is not needed, you could do like:
```java
jsonArray.stream().map(entry -> context.transform(entry, type)).forEach(resultFunction::accept);
```